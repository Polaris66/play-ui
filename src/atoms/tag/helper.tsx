import React from 'react';

import { getSplitMoveType } from 'helpers';
import Text from '../text';
import Icon from '../icon';

type Props = {
	value: string;
};

export function getTagValue({ value }: Props) {
	const splitMoveType = getSplitMoveType(value);

	const isMoveType = Array.isArray(splitMoveType);
	const isExport = value?.toLowerCase() === 'export';
	switch (true) {
		case isMoveType:
			return (
				<>
					<Text size="xSmall">{splitMoveType[0]}</Text>
					<Icon as="ArrowRight" size="Small" />
					<Text size="xSmall">{splitMoveType[1]}</Text>
				</>
			);
		case isExport:
			return (
				<>
					<Icon as="ArrowLeft" size="xSmall" />
					<Text size="xSmall">{value}</Text>
				</>
			);
		default:
			return <Text size="xSmall">{splitMoveType}</Text>;
	}
}
