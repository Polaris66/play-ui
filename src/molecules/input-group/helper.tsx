import { Icon, Toggle } from 'atoms';
import { IconsType } from 'types';

export const inputIcons = {
	second: [{ type: 'password', icon: 'EyesClose' }, ,],
};

type InputObjType = {
	type: string;
	icon: IconsType;
};

export function findIconName(data: any[], type: string) {
	const findObj = data?.find((item: InputObjType) => item?.type === type);

	return findObj?.icon || undefined;
}

type GetIconProps = {
	type: string;
	onShow?: () => void;
	show?: boolean;
};

export function getIcon({ type, onShow, show }: GetIconProps) {
	const iconsSecond = inputIcons.second;
	const icon = findIconName(iconsSecond, type);
	if (icon) {
		switch (type) {
			case 'password':
				return (
					<Toggle onClick={() => onShow && onShow()}>
						<Icon as={show ? 'Eyes' : icon} size="Medium" color="neutral400" />
					</Toggle>
				);
			default:
				return <Icon as={icon} size="Medium" color="neutral400" />;
		}
	}
}
