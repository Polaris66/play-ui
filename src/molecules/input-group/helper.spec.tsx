import { getIcon, inputIcons, findIconName } from './helper';

describe('helper functions', () => {
	it('getIcon()', () => {
		const icon = getIcon({ type: 'password' });

		expect(icon).toMatchSnapshot();
	});

	describe('findIconName()', () => {
		const cases = [
			[inputIcons.second, 'email', undefined],
			[inputIcons.second, 'password', 'Eyes'],
			[inputIcons.second, undefined, undefined],
			[undefined, undefined, undefined],
		];

		it.each(cases)(
			'given %p and %p as arguments, return %p',
			(firstArg: any, secondArg: any, expected) => {
				// TODO Types
				const result = findIconName(firstArg, secondArg);
				expect(result).toEqual(expected);
			}
		);
	});
});
