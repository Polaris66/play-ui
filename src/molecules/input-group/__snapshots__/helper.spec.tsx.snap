// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`helper functions getIcon() 1`] = `
<Toggle>
  <Icon
    as="Eyes"
    size="Medium"
  />
</Toggle>
`;
